### DISCUSSION TITLE
create vault token using var_sources

### DISCUSSION THREAD

**Discussion by schmurfy** [concourse/concourse#6150] [Help & Support]
I tried to play with `var_sources` but I don't understand how I can make this works.

I am running concourse directly on vms, the web node has a local vault already configured with auto_auth so I wanted my concourse pipeline to:
- create a temporary token which a short ttl
- use this token to create database credentials

My issue is that I don't see a way to create this token using `var_sources`, if I don't provide a auth method or token it raises an error (it could use the token I configured for global vault access) and if I use `auth_method` and `auth_params` it adds `/login` at the end when building the path where what I would need is `/create`.

Here is what I tried:
```hcl
  {
    name: "vault-pg",
    type: "vault",
    config: {
      url: "http://127.0.0.1:8200",
      path_prefix: "postgres/creds/",
      auth_backend: "token",
      auth_params: {
        "policy": "concourse",
        "role": "concourse-token"
      },
    },
  },
```

which leads to concourse failing querying `http://127.0.0.1:8200/v1/auth/token/login`

Is creating a token that way possible currently ?

**Comment by evanchaoli** [comment]
`var_sources` shares same Vault implementation as global Vault access, so in theory, if global vault works, `var_sources` should also work.

**Reply by schmurfy** [reply]
I saw that but for the global token what I ended up doing is using a wrapper script to start concourse and I issue the token myself passing it to concourse.

One way to solve this would be to be able to access the currently used vault token (configured by env) as a variable inside the pipeline config, for example: `((vault_token))` but I have seen nothing like that in the docs.

**Reply by schmurfy** [reply]
as a workaround what I did is that the wrapper script also write the generated token into vault, this feels stupid but it works for now xD
